docs/plugins.md

```diff
@@ -78,6 +78,16 @@ Install the plugin and configure the track id.
   window.$docsify = {
     ga: 'UA-XXXXX-Y',
   };
+  
+  // or multi gtag, the first element of the array must be a global site gtag id, each page should have only one global site tag instance. 
```

Comment:
This is confusing. Google's own terminology is confusing, so this isn't your fault. Let's simplify this for end users so they don't have to deal with Google's mess.

First, let's make sure we're using the correct terminology so people can easily locate and find more information on the IDs we're asking for. I'm not familiar with the term "global site gtag id", and when I search for it I do not find any results that match this  exact term (Google instead asks if I meant "global site tag id"). Are you referring to a GA3 "tracking ID" or a GA4 "measurement ID" here? If so, let's say this explicitly. I believe both of these IDs can be referred to as a "Google Property ID", but I'm not 100% clear on this.

Another idea is to remove the need for users to properly order their IDs and instead do the work for them to determine which ID should be used to initialize the `gtag` script. For example, instead of requiring users to list a GA4 or GA3 ID as the first item in the array, the plugin could search the array for the first instance of a preferred ID type. For example:

```js
window.$docsify = {
  ga: [
    'AW-XXXXXXXX', // Google Ads
    'DC-XXXXXXXX', // Floodlight
    'G-XXXXXXXX', // Google Analytics 4 (GA4)
    'UA-XXXXXXXX', // Google Universal Analytics (GA3)
  ],
};
```

1. Plugin searches array
2. Plugin finds GA4 "measurement ID" (`'G-XXXXXXXX'`)
3. Plugin passes GA4 ID to `initGlobalSiteTag()`
4. Plugin passes remaining IDs to `initAdditionalTag()`

The result is the same as if the end user made the GA4 ID the first item in the array, but by doing this work for them the user did not have to spend time trying to figure out the difference between a "Google Property ID", "Tracking ID", "Measurement ID", etc.

Thoughts?

Reply:
First, where does the requirement that "the first element of the array must be a global site gtag id" come from? I don't see this listed anywhere in Google's documentation or in places like StackOverflow:

   - https://developers.google.com/analytics/devguides/collection/gtagjs/
   - https://developers.google.com/tag-platform/gtagjs/install
   - https://stackoverflow.com/questions/46789189/google-analytics-gtag-multiple-analytics-account-tracking-ids

If this isn't a requirement, then we can remove the comment and process the IDs in the order they are listed (as you are now).

If there *is* a requirement for specific ID types to be listed first in the array, it's probably best to just identify those ID types specifically in the comment so it's easier for everyone to understand. For example:

```js
<script>
  // Single property
  window.$docsify = {
    ga: 'UA-XXXXX-Y',
  };

  // Multiple properties
  // IDs are processed in order the order listed
  // Analytics IDs should ordered before other IDs
  window.$docsify = {
    ga: [
      'G-XXXXXXXX', // Google Analytics 4 (GA4)
      'UA-XXXXXXXX', // Google Universal Analytics (GA3)
      'AW-XXXXXXXX', // Google Ads
      'DC-XXXXXXXX', // Floodlight
    ],
  };
</script>
```

This seems like an easier and more flexible solution that searching the array for specific ID types (which is what I recommended previously).

Thoughts?

Reply:
I understand, the current way is searching the array for specific ID types, which begin with `G-`.

Reply:
@ekoz --

Understood. But is this necessary? Where does the requirement come from that states specific types of IDs need to  go first?

I am not doubting you, but I'd like to understand it so we can make sure our code and documentation are as they should be.

Reply:
[https://developers.google.com/tag-platform/gtagjs/install](https://developers.google.com/tag-platform/gtagjs/install)

![image](https://user-images.githubusercontent.com/3312512/170906131-cf1d687f-991e-48f9-a447-ba9760ef4e9f.png)

Maybe I misunderstood this paragraph. I think a `global site tag` is needed here. Look at the Google demo, it may not be the `G-`tag. So, how to make the user specify this `global site tag`?

Reply:
@ekoz --

I believe the "global site tag" referred to in the documentation is the `<script>` tag that loads the `gtag` snippet:

```html
<script async src="https://www.googletagmanager.com/gtag/js?id=DC-ZZZZZZ"></script>
```

The documentation is saying that if you have multiple IDs, users **_SHOULD NOT_** configure them doing this:

```html
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXX"></script>
<script async src="https://www.googletagmanager.com/gtag/js?id=AW-YYYYYY"></script>
<script async src="https://www.googletagmanager.com/gtag/js?id=DC-ZZZZZZ"></script>
```

Instead, users should load the `gtag` snippet _once_, pass the first ID via a URL parameter to the snippet (e.g, `?id=G-XXXXXX`), then pass additional IDs using `gtag('configure', /* ID */)`.

Based on this, I don't think it matter which type of ID is passed as a URL parameter or via `gtag('configure')`. This will simplify things. We can remove the logic that searches for specific ID types and just process them in the order the appear in the array. The documentation can also be simplified:

```js
// Single ID
window.$docsify = {
  ga: 'UA-XXXXX-Y',
};

// Multiple IDs
window.$docsify = {
  ga: [
    'G-XXXXXXXX',  // Google Analytics 4 (GA4)
    'UA-XXXXXXXX', // Google Universal Analytics (GA3)
    'AW-XXXXXXXX', // Google Ads
    'DC-XXXXXXXX', // Floodlight
  ],
};
```

Reply:
ok, I think the data collector can be custom in `window.$docsify`, so that the user can define the data to be collected. If the user does not specify it, the default collector will be used. Do you think so?
```

window.$docsify = {
  ga: [
    'G-XXXXXXXX',  // Google Analytics 4 (GA4)
    'UA-XXXXXXXX', // Google Universal Analytics (GA3)
    'AW-XXXXXXXX', // Google Ads
    'DC-XXXXXXXX', // Floodlight
  ],

  // custom gaCollector
  gaCollector: function () {
    window.gtag('event', 'conversion', {
      send_to: 'AW-YYYYYY/AbC-D_efG-h12_34-567',
      value: 1.0,
      currency: 'USD',
    });

    // Measure Floodlight conversions
    window.gtag('event', 'conversion', {
      allow_custom_scripts: true,
      send_to: 'DC-ZZZZZZ/actions/locat304+standard',
    });
  },
};

// default ga collector
window.gtag('event', 'page_view', {
    page_title: document.title,
    page_location: location.href,
    page_path: location.pathname,
});
```